### ISSUE TITLE
Newlines inserted between pasted lines, after ssh into a remote linux system

### ISSUE THREAD

**Issue by vivekananthrp** [wez/wezterm#968]
I am using WezTerm in Windows 10 (Version 10.0.19042 Build 19042) with WSL2 Ubuntu 20.04.

Newlines are inserted between pasted lines copied from terminal. The issue happens only after ssh into a remote linux system.

I used "ctrl+shift+c" for copy and "ctrl+shift+v" for pasting. Any way to fix this?

https://user-images.githubusercontent.com/11735154/126307581-d6947785-67e8-42c0-8f21-ea1464cc16fa.mp4

_Originally posted by @vivekananthrp in https://github.com/wez/wezterm/issues/411#issuecomment-883278910_

**Comment by wez** [comment]
This is unfortunately expected behavior :-/

The Windows terminal wants pastes to have CRLF rather than just LF line endings.
If we don't use those then the pasting in WSL won't have any newlines at all.

wezterm will normalize to CRLF when it is running on Windows, but will normalize to LF
if the application has enabled bracketed paste mode.

Have you considered running `vim` instead of `vi`?  `vi` typically starts `vim` in a backwards
compatible mode that won't enable features like bracketed paste.  I have `vi` aliased to `vim`
because my fingers only want to type `vi` :-)

**Comment by vivekananthrp** [comment]
Hi, thanks for the reply. 

Using vim instead of vi didn't solve the issue.

I wonder why vim opened in local system has no issues, whereas the vim in the remote shell has this issue.

Even after using ":set paste" (in vim) before pasting, doesn't help. I still get extra newlines.

An interim solution is to run ":%s/^\n\(^\n\)*/\1/" (in vim) after every paste.
Ref: [Vim / Sed: Delete Only Single Blank Lines](https://stackoverflow.com/questions/34049711/vim-sed-delete-only-single-blank-lines)

**Comment by wez** [comment]
Try putting this into your .vimrc:

```
  let &t_BE="\<Esc>[?2004h"
  let &t_BD="\<Esc>[?2004l"
  let &t_PS="\<Esc>[200~"
  let &t_PE="\<Esc>[201~"
```

**Comment by ettoredn** [comment]
Im experiencing the opposite on macOS: pasted text is stripped of newlines and overwrites existing text in nano.

**Comment by wez** [comment]
@ettoredn please file a new issue of your own! That makes it easier to not let your comment fall through the cracks!
